tests/test_utils.py

```diff
@@ -163,3 +170,425 @@ def test_parse_log_components(ns_3_compiled_debug, config):
     with pytest.raises(ValueError):
         new_components = parse_log_components(log_components=log_components,
                                              ns3_log_components=ns3_log_components)
+
+
+def test_parse_logs():
+    data_dir = os.path.join(os.curdir, 'tests', str(time.strftime("%Y-%m-%d::%H-%M-%S")) + ':test_logs.json')
+
+    log_ls = ['+0.000000000s -1 [mac=00:00:00:00:00:00] FrameExchangeManager:SetWifiMac(0x5576595683e0, 0x557659603820)\n',
+              '+0.045510017s 1 [mac=00:00:00:00:00:01] FrameExchangeManager:RxStartIndication(0x5576595683e0, "PSDU reception started for ", +76us, " (txVector: ", txpwrlvl: 17 preamble: LONG channel width: 20 GI: 800 NTx: 97 Ness: 0 MPDU aggregation: 0 STBC: 0 FEC coding: BCC mode: OfdmRate6Mbps Nss: 1, ")")\n',
+              '+0.000000000s -1 WifiPhy:WifiPhy(0x557659571bd0)\n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber(): [DEBUG] Saving channel number configuration for initialization \n',
+              '+0.506390387s 1 [mac=00:00:00:00:00:01] FrameExchangeManager:StartTransmission(): [DEBUG] MPDU payload size=36, to=00:00:00:00:00:02, seq=16 \n',
+              'WifiPhy:AddStaticPhyEntity(HR/DSSS) \n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber(): [DEBUG] ()[]./-=\n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber() \n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber): [DEBUG] Saving channel number configuration for initialization \n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber(): DEBUG] Saving channel number configuration for initialization \n',
+              '+0.000000000s WifiPhy:SetChannelNumber(): [DEBUG] Saving channel number configuration for initialization \n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber(): Saving channel number configuration for initialization \n',
+              '+0.000000000s -1 WifiPhy:SetChannelNumber():  \n'
+              ]
```

Comment:
other suggested tests:
* a log with severity INFO (it should find some spaces between the square brackets, check that they are handled correctly
* also check extended context with node id instead of mac
* check what happens if the log message contains escape characters such as `:`

I think you already covered most corner cases yourself, good job :)

Reply:
note: you are reusing these logs in other test functions as well, you might want to make this a global variable (i.e., put it outside the function at the beginning of the test file so that all functions can reach it).
same goes for expected_list and similar verbose variables

Reply:
> note: you are reusing these logs in other test functions as well, you might want to make this a global variable (i.e., put it outside the function at the beginning of the test file so that all functions can reach it).
same goes for expected_list and similar verbose variables

Corrected in c8e8db3.

Reply:
> other suggested tests

Corrected in [3dc3eba](https://github.com/signetlabdei/sem/pull/51/commits/3dc3eba078392844de0531e4f4ace7e5336bf241).